os.ErrNoDeadline is undefined

While trying to compile https://github.com/cockroachdb/cockroachdb-parser for WASI I run into the following error:

```
# github.com/cockroachdb/errors/errutil
../../go/pkg/mod/github.com/cockroachdb/errors@v1.9.0/errutil/format_error_special.go:42:6: undefined: os.ErrNoDeadline
```

It seems like `os.ErrNoDeadline` is just missing from os package and it should be trivial to add it there, similar to how `os.ErrDeadlineExceeded` is backported from `poll` package. I will make a PR.

--- a/src/os/errors.go
+++ b/src/os/errors.go
@@ -32,6 +32,10 @@ var (
 // The following code is copied from the official implementation.
 // src/internal/poll/fd.go
 
+// ErrNoDeadline is returned when a request is made to set a deadline
+// on a file type that does not use the poller.
+var ErrNoDeadline = errors.New("file type does not support deadline")
+
 // ErrDeadlineExceeded is returned for an expired deadline.
 // This is exported by the os package as os.ErrDeadlineExceeded.
 var ErrDeadlineExceeded error = &DeadlineExceededError{}

